diff --git a/lib/jekyll/drops/rdf_resource.rb b/lib/jekyll/drops/rdf_resource.rb
index ab782de..3b76919 100644
--- a/lib/jekyll/drops/rdf_resource.rb
+++ b/lib/jekyll/drops/rdf_resource.rb
@@ -252,8 +252,8 @@ module Jekyll #:nodoc:
         #
         def generate_file_name()
           fragment_holder = nil
-          domain_name = URI::split(Jekyll::JekyllRdf::Helper::RdfHelper::config["url"])[2].to_s
-          baseurl = Jekyll::JekyllRdf::Helper::RdfHelper::config["baseurl"].to_s
+          domain_name = URI::split(Jekyll::JekyllRdf::Helper::RdfHelper::domainiri)[2].to_s
+          baseurl = Jekyll::JekyllRdf::Helper::RdfHelper::pathiri.to_s
           (("/".eql? baseurl[-1]) || (baseurl.empty? && ("/".eql? domain_name[-1]))) ? rdfsites="rdfsites/": rdfsites="/rdfsites/"
           if(term.to_s[0..1].eql? "_:")
             file_name = "#{rdfsites}blanknode/#{term.to_s.gsub('_:','blanknode_')}/" # ':' can not be used on windows
diff --git a/lib/jekyll/helper/rdf_general_helper.rb b/lib/jekyll/helper/rdf_general_helper.rb
index 39331fa..db8bbf7 100644
--- a/lib/jekyll/helper/rdf_general_helper.rb
+++ b/lib/jekyll/helper/rdf_general_helper.rb
@@ -80,14 +80,6 @@ module Jekyll
           end
         end
 
-        def self.config= config
-          @@config = config
-        end
-
-        def self.config
-          @@config
-        end
-
         def self.prefixes
           if(@@usePage)
             return @@page.data
@@ -95,6 +87,22 @@ module Jekyll
             return @@prefixes
           end
         end
+
+        def self.domainiri= domain
+          @@domainiri = domain
+        end
+
+        def self.domainiri
+          @@domainiri
+        end
+
+        def self.pathiri= path
+          @@baseiri = path
+        end
+
+        def self.pathiri
+          @@baseiri
+        end
       end
 
     end
diff --git a/lib/jekyll/helper/rdf_generator_helper.rb b/lib/jekyll/helper/rdf_generator_helper.rb
index 20b7558..b038dee 100644
--- a/lib/jekyll/helper/rdf_generator_helper.rb
+++ b/lib/jekyll/helper/rdf_generator_helper.rb
@@ -53,13 +53,25 @@ module Jekyll
           end
 
           @global_config = Jekyll.configuration({})
-
           #small fix because global_config doesn't work in a test enviorment
           if(!@global_config.key? "url")
             @global_config["url"] = site.config["url"]
             @global_config["baseurl"] = site.config["baseurl"]
           end
-          Jekyll::JekyllRdf::Helper::RdfHelper::config = @global_config
+
+          if(@config["baseiri"].nil?)
+            Jekyll::JekyllRdf::Helper::RdfHelper::domainiri = @global_config["url"]
+            Jekyll::JekyllRdf::Helper::RdfHelper::pathiri = @global_config["baseurl"]
+          else
+            uri = Addressable::URI.parse(@config["baseiri"]).to_hash
+            domainuri = ""
+            domainuri << "#{uri[:scheme]}://" unless uri[:scheme].nil?
+            domainuri << "#{uri[:userinfo]}@" unless uri[:userinfo].nil?
+            domainuri << "#{uri[:host]}" unless uri[:host].nil?
+            domainuri << ":#{uri[:port]}" unless uri[:port].nil?
+            Jekyll::JekyllRdf::Helper::RdfHelper::domainiri = domainuri
+            Jekyll::JekyllRdf::Helper::RdfHelper::pathiri = uri[:path][0..-1].to_s unless uri[:path].nil?
+          end
           return true
         end
         ##
diff --git a/test/test_rdf_main_generator.rb b/test/test_rdf_main_generator.rb
index c485676..770b271 100644
--- a/test/test_rdf_main_generator.rb
+++ b/test/test_rdf_main_generator.rb
@@ -15,6 +15,15 @@ class TestRdfTemplateMapper < Test::Unit::TestCase
       assert_equal("/INF3580", @global_config["baseurl"])
     end
 
+    should "load baseiri if available" do
+      Jekyll.logger.warn "enter"
+      test_config = Jekyll.configuration({"url" => "http://www.ifi.uio.no", "baseurl" => "/INF3580", "jekyll_rdf" => {"baseiri" => "http://www.example.org/path/to/our/special/resource"}})
+      site = Jekyll::Site.new(test_config)
+      load_config(site)
+      assert_equal "http://www.example.org", Jekyll::JekyllRdf::Helper::RdfHelper::domainiri
+      assert_equal "/path/to/our/special/resource", Jekyll::JekyllRdf::Helper::RdfHelper::pathiri
+    end
+
     should "fail if jekyll-rdf is included but not configured" do
       TestHelper::setErrOutput
       @site = res_helper.create_bad_fetch_site()
diff --git a/test/test_rdf_page_data.rb b/test/test_rdf_page_data.rb
index a649072..495f532 100644
--- a/test/test_rdf_page_data.rb
+++ b/test/test_rdf_page_data.rb
@@ -129,7 +129,6 @@ class TestRdfTemplateMapper < Test::Unit::TestCase
     end
 
     should "should create pages with page.name and page.dir reflecting their resources iri" do
-      Jekyll::JekyllRdf::Helper::RdfHelper::config = Jekyll.configuration({'url' => "http://ex.org", 'baseurl' => "/blog" })
       config = Jekyll.configuration(TestHelper::TEST_OPTIONS)
       site = Jekyll::Site.new(config)
       site.data['resources'] = []
@@ -145,10 +144,10 @@ class TestRdfTemplateMapper < Test::Unit::TestCase
     end
   end
 
-  context "Jekyll::JekyllRdf::Drops::RdfResource.render_path with empty baseurl"do
+  context "Jekyll::JekyllRdf::Drops::RdfResource.render_path with empty baseurl" do
     setup do
-      Jekyll::JekyllRdf::Helper::RdfHelper::config = Jekyll.configuration({'url' => "http://ex.org", 'baseurl' => "" })
-
+      Jekyll::JekyllRdf::Helper::RdfHelper::domainiri = "http://ex.org"
+      Jekyll::JekyllRdf::Helper::RdfHelper::pathiri = ""
     end
 
     should "correctly render simple urls" do
@@ -174,7 +173,8 @@ class TestRdfTemplateMapper < Test::Unit::TestCase
 
   context "Jekyll::JekyllRdf::Drops::RdfResource.render_path with '/' as baseurl"do
     setup do
-      Jekyll::JekyllRdf::Helper::RdfHelper::config = Jekyll.configuration({'url' => "http://ex.org", 'baseurl' => "/" })
+      Jekyll::JekyllRdf::Helper::RdfHelper::domainiri = "http://ex.org"
+      Jekyll::JekyllRdf::Helper::RdfHelper::pathiri = "/"
     end
 
     should "correctly render simple urls" do
@@ -200,7 +200,8 @@ class TestRdfTemplateMapper < Test::Unit::TestCase
 
   context "Jekyll::JekyllRdf::Drops::RdfResource.render_path with subdirectory baseurl"do
     setup do
-      Jekyll::JekyllRdf::Helper::RdfHelper::config = Jekyll.configuration({'url' => "http://ex.org", 'baseurl' => "/blog" })
+      Jekyll::JekyllRdf::Helper::RdfHelper::domainiri = "http://ex.org"
+      Jekyll::JekyllRdf::Helper::RdfHelper::pathiri = "/blog"
     end
 
     should "correctly render simple urls" do
@@ -230,7 +231,8 @@ class TestRdfTemplateMapper < Test::Unit::TestCase
 
   context "Jekyll::JekyllRdf::Drops::RdfResource.render_path with subdirectory baseurl ending with slash"do
     setup do
-      Jekyll::JekyllRdf::Helper::RdfHelper::config = Jekyll.configuration({'url' => "http://ex.org", 'baseurl' => "/blog/" })
+      Jekyll::JekyllRdf::Helper::RdfHelper::domainiri = "http://ex.org"
+      Jekyll::JekyllRdf::Helper::RdfHelper::pathiri = "/blog/"
     end
 
     should "correctly render simple urls" do
@@ -259,37 +261,33 @@ class TestRdfTemplateMapper < Test::Unit::TestCase
   end
 
   context "RdfResource" do
-    setup do
-      Jekyll::JekyllRdf::Helper::RdfHelper::config = Jekyll.configuration({'url' => "http://ex.org", 'baseurl' => '/blog/'})
-    end
-
     should "correctly disect its iri into file name and file directory" do
       resource = Jekyll::JekyllRdf::Drops::RdfResource.new("http://ex.org/blog/bla/a")
-      Jekyll::JekyllRdf::Helper::RdfHelper::config['url'] = "http://ex.org"
-      Jekyll::JekyllRdf::Helper::RdfHelper::config['baseurl'] = "/blog/"
+      Jekyll::JekyllRdf::Helper::RdfHelper::domainiri = "http://ex.org"
+      Jekyll::JekyllRdf::Helper::RdfHelper::pathiri = "/blog/"
       assert_equal "a.html", resource.filename
       assert_equal "bla/", resource.filedir
       resource = Jekyll::JekyllRdf::Drops::RdfResource.new("http://ex.org/blog/bla/a")
-      Jekyll::JekyllRdf::Helper::RdfHelper::config['url'] = "http://ex.org"
-      Jekyll::JekyllRdf::Helper::RdfHelper::config['baseurl'] = ""
+      Jekyll::JekyllRdf::Helper::RdfHelper::domainiri = "http://ex.org"
+      Jekyll::JekyllRdf::Helper::RdfHelper::pathiri = ""
       assert_equal "a.html", resource.filename
       assert_equal "/blog/bla/", resource.filedir
     end
 
     should "set the filedir to rdfsites/... if the site url and baseurl coincides with the resource iri" do
       resource = Jekyll::JekyllRdf::Drops::RdfResource.new("http://ex.org/blog/bla/a")
-      Jekyll::JekyllRdf::Helper::RdfHelper::config['url'] = ""
-      Jekyll::JekyllRdf::Helper::RdfHelper::config['baseurl'] = ""
+      Jekyll::JekyllRdf::Helper::RdfHelper::domainiri = ""
+      Jekyll::JekyllRdf::Helper::RdfHelper::pathiri = ""
       assert_equal "a.html", resource.filename
       assert_equal "/rdfsites/http/ex.org/blog/bla/", resource.filedir
       resource = Jekyll::JekyllRdf::Drops::RdfResource.new("http://ex.org/blog/bla/a")
-      Jekyll::JekyllRdf::Helper::RdfHelper::config['url'] = "http://ex.org"
-      Jekyll::JekyllRdf::Helper::RdfHelper::config['baseurl'] = "t"
+      Jekyll::JekyllRdf::Helper::RdfHelper::domainiri = "http://ex.org"
+      Jekyll::JekyllRdf::Helper::RdfHelper::pathiri = "t"
       assert_equal "a.html", resource.filename
       assert_equal "/rdfsites/http/ex.org/blog/bla/", resource.filedir
       resource = Jekyll::JekyllRdf::Drops::RdfResource.new("http://ex.org/blog/bla/a")
-      Jekyll::JekyllRdf::Helper::RdfHelper::config['url'] = "http://ex.org"
-      Jekyll::JekyllRdf::Helper::RdfHelper::config['baseurl'] = "/blog/s"
+      Jekyll::JekyllRdf::Helper::RdfHelper::domainiri = "http://ex.org"
+      Jekyll::JekyllRdf::Helper::RdfHelper::pathiri = "/blog/s"
       assert_equal "a.html", resource.filename
       assert_equal "/rdfsites/http/ex.org/blog/bla/", resource.filedir
     end
